### PR TITLE
[MOJ-3593] Added Token Gating API

### PR DESCRIPTION
**Pull request by sakthi-ionixx**
## Related Issue
https://mojitoinc.atlassian.net/browse/MOJ-3593

## Changes

1. Added Token Gating Group API's checkTokenGating, createTokenGatingRule
2. Update Readme for Token Gating API's

## Screenshots

Screenshot from 2023-08-02 17-41-51

## Checklist

- [x] param and service should be meaningful(name should indicate, why it is used rather than which service its calling.)
- [x] Param should be type and description in ReadMe
- [x] don't use arrow function functional components useCallback
- [x] Always try to use const. There is a very good reason to use let
- [x] Should be exported only Hooks and interface
- [x] Before pushing to GitHub delete all unused states, spacing, and console.log, if exist
- [x] PR should contain  screenshots Unit test with 100% coverage
- [x] Lint should be passed on to PR

### CODE REVIEW

File: src/TokenGating/TokenGatingParam.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+interface GatingRule {
+    ruleType: RuleType;
+    contractAddress?: ContractAddressGating;
+    contractAddressWithTokenId?: ContractAddressWithTokenIDGating;
+    metaData?: MetaDataGatingInput;
+    onchain?: OnchainGating;
+}
+
+interface gatingInput {
```
Comment (kameshkishoreIonixx) [inline src/TokenGating/TokenGatingParam.ts:42]: interface name should be capitalise , GatingInput

File: README.md [hunk 2]
```diff
@@ -2115,6 +2115,172 @@ import { useOffer, SubmitProofOfOfferData } from "@mojitoinc/mojito-core-service
...
+1. [createTokenGating](#createTokenGating)
+2. [checkTokenGating](#checkTokenGating)
+
+
+### createTokenGating
+
+  This API is used to create a Token Gating for an NFT.
+
+```
+import { useTokenGating, CreateTokenGatingData } from "@mojitoinc/mojito-core-service";
```
Comment (kameshkishoreIonixx) [inline README.md:2130]: can we change the import name as "@mojito-inc/core-service" for all the imports in read me

Comment (sakthi-ionixx) [reply README.md]: changes updated

File: src/TokenGating/TokenGatingParam.ts (hunk 1 above)
Comment (sakthi-ionixx) [reply src/TokenGating/TokenGatingParam.ts]: Changed to Capitalize

**Review by kameshkishoreIonixx — APPROVED** [review]

**Pull request merged**